compute/cloud-client/src/main/java/compute/ListImages.java

```diff
@@ -0,0 +1,102 @@
+
+    // page_size: size of the pages you want the API to return on each call.
+    int pageSize = 2;
+    listImagesByPage(project, pageSize);
+  }
+
+  // [START compute_images_list]
+  // Prints a list of all non-deprecated image names available in given project.
+  public static void listImages(String project) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `instancesClient.close()` method on the client to
+    // safely clean up any remaining background resources.
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+
+      // Listing only non-deprecated images to reduce the size of the reply.
+      ListImagesRequest imagesRequest = ListImagesRequest.newBuilder()
+          .setProject(project)
+          .setMaxResults(2)
+          .setFilter("deprecated.state != DEPRECATED")
+          .build();
+
+      // Although the `setMaxResults` parameter is specified in the request, the iterable returned
+      // by the `list()` method hides the pagination mechanic. The library makes multiple
+      // requests to the API for you, so you can simply iterate over all the images.
+      int imageCount = 0;
+      for (Image image : imagesClient.list(imagesRequest).iterateAll()) {
+        imageCount++;
+        System.out.println(image.getName());
+      }
+      System.out.println(String.format("Image count in %s is: %s", project, imageCount));
+    }
+  }
+  // [END compute_images_list]
+
+  // [START compute_images_list_page]
+  //  Prints a list of all non-deprecated image names available in a given project,
+  //  divided into pages as returned by the Compute Engine API.
+  public static void listImagesByPage(String project, int pageSize) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `instancesClient.close()` method on the client to
+    // safely clean up any remaining background resources.
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+
+      // Listing only non-deprecated images to reduce the size of the reply.
+      ListImagesRequest imagesRequest = ListImagesRequest.newBuilder()
+          .setProject(project)
+          .setMaxResults(pageSize)
+          .setFilter("deprecated.state != DEPRECATED")
+          .build();
+
+      // Use the `iteratePages` attribute of returned iterable to have more granular control of
+      // iteration over paginated results from the API. Each time you want to access the
+      // next page, the library retrieves that page from the API.
+      int pageNumber = 1;
+      for (ListPage page : imagesClient.list(imagesRequest).iteratePages()) {
+        System.out.println("Page Number: " + pageNumber++);
```

Comment:
Shouldn't we be incrementing after this?

Reply:
Yes, I am incrementing the pageNumber as part of the sysout statement!